provider/azure/environ.go

```diff
@@ -1292,46 +1327,78 @@ func (env *azureEnviron) deleteVirtualMachine(
 	for _, nic := range networkInterfaces {
 		nicName := to.String(nic.Name)
 		logger.Tracef("deleting NIC %q", nicName)
-		resultCh, errCh := nicClient.Delete(env.resourceGroup, nicName, nil)
-		if result, err := <-resultCh, <-errCh; err != nil {
+		nicFuture, err := nicClient.Delete(sdkCtx, env.resourceGroup, nicName)
+		if err != nil {
 			msg := "deleting NIC"
 			if errorutils.MaybeInvalidateCredential(err, ctx) {
 				return invalidatedCredential(err, msg)
 			}
-			if !isNotFoundResponse(result) {
+			if !isNotFoundResponse(autorest.Response{nicFuture.Response()}) {
```

Comment:
I wonder we need both a isNotFoundResponse to take a http.Response and a isNotFoundResult to take the autorest.Response from the Result() call and have the isNotFoundResponse wrap and call isNotFoundResult.
Avoids all this struct boilerplate